File: 88-merge-sorted-array/merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& arr1, int m, vector<int>& arr2, int n) {
        int i = m - 1;           // Last index of valid elements in arr1
        int j = n - 1;           // Last index of arr2
        int k = m + n - 1;       // Last index of full arr1

        // Merge from back to front
        while (i >= 0 && j >= 0) {
            if (arr1[i] > arr2[j]) {
                arr1[k--] = arr1[i--];
            } else {
                arr1[k--] = arr2[j--];
            }
        }

        // Copy remaining elements from arr2 (if any)
        while (j >= 0) {
            arr1[k--] = arr2[j--];
        }
    }
};

File: 88-merge-sorted-array/merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        ArrayList<Integer> arr1 = new ArrayList<>();
        for (int i=0;i<m;i++){
            arr1.add(nums1[i]);
        }
        for (int i=0;i<n;i++){
            arr1.add(nums2[i]);
        }
        Collections.sort(arr1);
        for(int i=0;i<arr1.size();i++){
            nums1[i]= arr1.get(i);
        }
        
    }
}